en/includes/deploy/security/keystores/configure-custom-keystores.md

```diff
@@ -0,0 +1,125 @@
+# Configure Custom KeyStores for Authentication Protocols
+
+By default, WSO2 Identity Server uses the private key and certificate from the Primary KeyStore in all authentication protocols such as,
+
+- Sign JWT tokens in OAuth authentication flows  
+- Verify JWT tokens signed by WSO2 identity server  
+- Sign SAML assertions in SAML authentication flows
+
+WSO2 Identity Server also allows configuring separate keystores for each authentication protocol.
+
+!!! note
+    Currently only the following authentication protocols are supported,
+
+    - OAuth 2.0
+    - WS-Federation
+    - WS-Trust (Needs to be added via the WS-Trust connector)
+    - SAML
+
+This document provides step-by-step instructions for configuring a custom keystore for a specific authentication protocol. You can define multiple custom keystores and assign a separate keystore to each protocol for better management and security.
+
+## Configure Custom KeyStore for OAuth, WS-Trust or WS-Federation
+
+1. Place your custom key store file in `<IS-HOME>/repository/resources/security/` directory.  
+2. Add the following configuration to `deployment.toml` file.
+
+    === "JKS"
+        ``` toml
+        [[keystore.custom]]
+        file_name = "<keystore name>"
+        password = "<password>"
+        type = "JKS"
+        alias = "<private key alias>"
+        key_password = "<password>"
+        ```
+
+    === "PKCS12"
+        ``` toml
+        [[keystore.custom]]
+        file_name = "<keystore name>"
+        password = "<password>"
+        type = "PKCS12"
+        alias = "<private key alias>"
+        key_password = "<password>"
+        ```
+
+    ### Configuration Parameters,
+
+    | Configuration | Details |
+    | :---- | :---- |
+    | `file_name` | Key store file name. |
+    | `password` | Key Store password. |
+    | `type` | Key Store Type. |
+    | `alias` | Alias value of the private key. |
+    | `key_password` | Private key password. |
```

Comment:
shall we align this to the left most side?

Reply:
This couldn't be done because the section is defined under a numbered point, and making this change would alter the layout.

Reply:
I removed the heading and added it as regular text because a heading is not appropriate in this context.

Reply:
Ack.